src/main/java/net/sf/jabref/gui/help/FXAlert.java

```diff
@@ -0,0 +1,116 @@
+ * windows in the java fx thread.
+ * <p>To create a custom java fx dialog one should not extend this class but rather
+ * create the layout of the dialog as a fxml file, load it via the {@link FXMLLoader}
+ * and set it as dialog pane through the {@link setDialogPane(DialogPane)} methode of
+ * the FXAlert object.
+ *
+ */
+public class FXAlert extends Alert {
+
+    /**
+     * The WindowAdapter will be added to all swing windows once an instance
+     * of this class is shown and redirects the focus towards this instance.
+     * It will be removed once the instance of this class gets hidden.
+     *
+     */
+    private final WindowAdapter fxOverSwingHelper = new WindowAdapter() {
+
+        @Override
+        public void windowActivated(WindowEvent e) {
+            Platform.runLater(() -> {
+                Stage fxDialogWindow = (Stage) getDialogPane().getScene().getWindow();
+                fxDialogWindow.toFront();
+                fxDialogWindow.requestFocus();
+            });
+        }
+
+        @Override
+        public void windowGainedFocus(WindowEvent e) {
+            Platform.runLater(() -> {
+                Stage fxDialogWindow = (Stage) getDialogPane().getScene().getWindow();
+                fxDialogWindow.toFront();
+                fxDialogWindow.requestFocus();
+            });
+        }
+    };
+
+
+    public FXAlert(AlertType type, String title, Image image) {
+        this(type, title);
+        Stage fxDialogWindow = (Stage) getDialogPane().getScene().getWindow();
+        fxDialogWindow.getIcons().add(image);
+    }
+
+    public FXAlert(AlertType type, String title) {
+        super(type);
+        setTitle(title);
+        Stage fxDialogWindow = (Stage) getDialogPane().getScene().getWindow();
+
+        fxDialogWindow.setOnShown(evt -> {
+            setSwingWindowsEnabledAndFocusable(false);
+            setLocationRelativeToMainWindow();
+        });
+
+        fxDialogWindow.setOnHiding(evt -> setSwingWindowsEnabledAndFocusable(true));
+
+        fxDialogWindow.setOnCloseRequest(evt -> this.close());
+    }
+
+    public void setDialogStyle(String pathToStyleSheet) {
+        getDialogPane().getScene().getStylesheets().add(pathToStyleSheet);
+    }
+
+    private void setSwingWindowsEnabledAndFocusable(boolean enabled) {
+        for (Window swingWindow : Window.getWindows()) {
+            swingWindow.setEnabled(enabled);
+            if (!enabled) {
+                swingWindow.addWindowListener(fxOverSwingHelper);
+            } else {
+                swingWindow.removeWindowListener(fxOverSwingHelper);
+            }
+        }
+    }
+
+    private void setLocationRelativeToMainWindow() {
+        double mainX = JabRef.jrf.getLocationOnScreen().getX();
+        double mainY = JabRef.jrf.getLocationOnScreen().getY();
+        double mainWidth = JabRef.jrf.getSize().getWidth();
+        double mainHeight = JabRef.jrf.getSize().getHeight();
+
```

Comment:
mainWidth is not such a meaningful name. I think that mainWindowWidth for example would be better